components/count-badge/count-badge-mixin.js

```diff
@@ -0,0 +1,185 @@
+}
+
+:host([size="small"]) .d2l-count-badge-number {
+	border-radius: 0.55rem;
+	font-size: 0.6rem;
+	line-height: 0.9rem;
+	padding-left: 0.3rem;
+	padding-right: 0.3rem;
+}
+
+:host([size="large"]) .d2l-count-badge-number {
+	border-radius: 0.7rem;
+	font-size: 0.8rem;
+	line-height: 1.2rem;
+	padding-left: 0.4rem;
+	padding-right: 0.4rem;
+}
+`];
+
+export const CountBadgeMixin = superclass => class extends LocalizeCoreElement(RtlMixin(superclass)) {
+
+	static get properties() {
+		return {
+			/**
+			 * Optionally choose to announce changes to the badge with an aria-live region. If the number property is changed, the text will be read by screenreaders. Defaults to false.
+			 * @type {boolean}
+			 */
+			announceChanges: {
+				type: Boolean,
+				attribute: 'announce-changes'
+			},
+			/**
+			 * Optionally add a tooltip on the badge. Defaults to false.
+			 * @type {boolean}
+			 */
+			hasTooltip: {
+				type: Boolean,
+				attribute: 'has-tooltip'
+			},
+			/**
+			 * Optionally choose to not render the count badge when the number is zero. Defaults to false.
+			 * @type {boolean}
+			 */
+			hideZero: {
+				type: Boolean,
+				attribute: 'hide-zero'
+			},
+			/**
+			 * Optionally specify a digit limit, after which numbers are truncated. Defaults to two for "notification" type and no limit for "count" type.
+			 * @type {number}
+			 */
+			maxDigits: {
+				type: Number,
+				attribute: 'max-digits'
+			},
+			/**
+			 * The number to be displayed on the badge. Must be a positive integer.
+			 * @type {number}
+			 */
+			number: {
+				type: Number,
+				attribute: 'number'
+			},
+			/**
+			 * The size of the badge. Defaults to "small".
+			 * @type {'small'|'large'}
+			 */
+			size: {
+				type: String,
+				reflect: true,
+				attribute: 'size'
+			},
+			/**
+			 * Optionally choose to add a tab stop to the badge. Defaults to false.
+			 * @type {boolean}
+			 */
+			tabStop: {
+				type: Boolean,
+				attribute: 'tab-stop'
+			},
+			/**
+			 * * Descriptive text for the badge which will act as an accessible label and tooltip text when tooltips are enabled.
+			 * @type {string}
+			 */
+			text: {
+				type: String
+			},
+			/**
+			 * The type of the badge. Defaults to "count".
+			 * @type {'count'|'notification'}
+			 */
+			type: {
+				type: String,
+				reflect: true,
+				attribute: 'type'
+			}
+		};
+	}
+
+	constructor() {
+		super();
+		this.announceChanges = false;
+		this.hasTooltip = false;
+		this.hideZero = false;
+		this.size = 'small';
+		this.tabStop = false;
+		this.text = '';
+		this.type = 'count';
+		this._textId = getUniqueId();
+	}
+
+	connectedCallback() {
+		super.connectedCallback();
+		if (!this.maxDigits && this.type === 'notification') {
+			// default to two digits for notification type
+			this.maxDigits = 2;
+		}
+	}
+
+	getOffscreenId() {
+		return this.hasTooltip ? undefined : this._textId;
+	}
+
+	renderCount(numberStyles) {
+		let numberString = `${this.number}`;
```

Comment:
Wondering if you want to put the number through [formatNumber](https://github.com/BrightspaceUI/intl#number-formatting)? That way "1000000" would render as "1,000,000" (in most English locales anyway).

Reply:
Yep, I had this as a separate story to investigate. Great that there's already a helper to do it. 🎊